frontend/controller/sw-namespace.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
```

Comment:
What is this file? Is it supposed to run in the service worker? If so, shouldn't it be with the other `sw-*` files under the `serviceworkers/` directory?

(Also maybe a comment at the top here explaining this..)

Reply:
> Is it supposed to run in the service worker?

Yes